### PR TITLE
refactor(clickhouse): use more sqlglot constructs

### PR DESCRIPTION
**Pull request by cpcloud**
Add a few more sqlglot conveniences for codegen and default to the MergeTree engine for file reading.

### CODE REVIEW

File: ibis/backends/base/sqlglot/__init__.py [hunk 1]
```diff
@@ -52,11 +52,30 @@ def exists(self, query):
...
 def lit(val):
     return sg.exp.Literal(this=str(val), is_string=isinstance(val, str))
 
 
+def interval(value, *, unit):
+    return sg.exp.Interval(this=_to_sqlglot(value), unit=sg.exp.var(unit))
+
+
+f = FuncGen()
+c = ColGen()
```
Comment (jcrist) [inline ibis/backends/base/sqlglot/__init__.py:78]: Single letter lowercase names intended to be imported by other modules feels weird to me. How about calling these `FUNCS` and `COLS` instead? Or `F` and `C` at least?

Comment (cpcloud) [reply ibis/backends/base/sqlglot/__init__.py:78]: `F` and `C` sounds good to me. Definitely want to keep them easy to type.

Comment (cpcloud) [reply ibis/backends/base/sqlglot/__init__.py]: Fixed!

**Review by jcrist — APPROVED** [review]

**Comment by jcrist** [comment]
Looks like you have a small conflict, otherwise LGTM!

**Comment by cpcloud** [comment]
I'm going to revert the `sign` change in case anyone is using ibis with an older clickhouse version.

**Pull request merged**